src/modules/navigator/navigator_main.cpp

```diff
@@ -563,6 +563,22 @@ Navigator::run()
 
 					break;
 
+				case RTL::RTL_CLOSEST:
+					if (rtl_activated) {
+						mavlink_and_console_log_info(get_mavlink_log_pub(), "RTL Closest landing point activated");
+					}
+
+					// if RTL is set to use a mission landing and mission has a planned landing, then use MISSION to fly there directly
+					if (on_mission_landing() && !get_land_detected()->landed) {
```

Comment:
Shouldn't we let RTL handle this case as well?

Reply:
no don't think so, as in this case we're already in the RTL using mission landing (it set's navigation_mode_new to rtl in the first pass, but to _mission when the RTL using mission landing has startted)

Reply:
Ok, looking at `on_mission_landing()`, that looks right then. Can you merge the code with `case RTL::RTL_LAND:`? It's the same up to the message.

Reply:
done